docs/reference/ingest/processors/data-stream-router.asciidoc

```diff
@@ -0,0 +1,43 @@
+[[data-stream-router-processor]]
+=== Data stream router processor
+++++
+<titleabbrev>Data stream router</titleabbrev>
+++++
+
+The `data_stream_router` processor allows to route a document from one data stream to another data stream.
+It can use both static values or values from the document to determine the target data stream.
+
+The name of a data stream is comprised of three parts and looks like this: `<type>-<dataset>-<namespace>`.
+See the {fleet-guide}/data-streams.html#data-streams-naming-scheme[data stream naming scheme] documentation for more details.
+
+NOTE: `data_stream_router` processor can only be used on data streams that follow the data streams naming scheme.
+Trying to use this processor on a data stream with a non-compliant name will raise an exception.
+
+After a `data_stream_router` processor has been executed, all the other processors of the current pipeline are skipped.
+This means that at most one `data_stream_router` processor is ever executed within a pipeline,
+allowing to define mutually exclusive routing conditions,
+similar to a if, else-if, else-if, … condition.
+
+[[data-stream-router-options]]
+.Data stream router options
+[options="header"]
+|======
+| Name        | Required  | Default  | Description
+| `dataset`   | no        | -        | A static value for the dataset part of the data stream name. In addition to the criteria for <<indices-create-api-path-params, index names>>, cannot contain `-` and must be no longer than 100 characters. Example values are `nginx.access` and `nginx.error`. If not set, gets the value of the field `data_stream.dataset` from the document. When using values from the document, the processor replaces invalid characters with `_`. If the option is not set and the document also doesn't contain a corresponding field, it uses the `<dataset>` part of the index name as a fallback.
+| `namespace` | no        | -        | A static value for the namespace part of the data stream name. See the criteria for <<indices-create-api-path-params, index names>> for allowed characters. Must be no longer than 100 characters. If not set, gets the value of the field `data_stream.namespace` from the document. When using values from the document, the processor replaces invalid characters with `_`. If the option is not set and the document also doesn't contain a corresponding field, it uses the `<namespace>` part of the index name as a fallback.
+include::common-options.asciidoc[]
+|======
+
+NOTE: It's not possible to change the `type` of the data stream by setting the `data_stream.type` in the document.
+
```

Comment:
What will happen if the user does not have permissions for the target index? We should verify that a good error is returned for the rejection (I assume doc is rejected)

Reply:
You'll get the same error as trying to directly index into a data stream you don't have permissions for. This is how the error would look like:

```json
{"type":"security_exception","reason":"action [indices:admin/auto_create] is unauthorized for API key id [8-dt9H8BqGblnY2uSI--] of user [elastic/fleet-server] on indices [logs-foo-default], this action is granted by the index privileges [auto_configure,create_index,manage,all]"}
```